TheMovieDb/Home/Contoller/HomeViewController.swift

```diff
@@ -7,21 +7,226 @@
-        // Do any additional setup after loading the view.
+        configureUICollection()
+        fetchDataAPI()
+    }
+    // MARK: - API
+    private func reloadCollectionView() {
+        DispatchQueue.main.async {
+            self.collectionView.reloadData()
+        }
+    }
+    
+    private func fetchDataAPI() {
+        fetchData(typeMovieSection: .popular)
+        fetchData(typeMovieSection: .upcoming)
+        fetchData(typeMovieSection: .topRated)
+        fetchData(typeMovieSection: .playingNow)
+        fetchData(typeMovieSection: .trending)
+        group.notify(queue: .main) {
+            self.reloadCollectionView()
+        }
+    }
+    
+    private func fetchData(typeMovieSection: MovieGroupSections) {
+        group.enter()
+        MovieAPI.shared.fetchData(endPoint: typeMovieSection.path, completion: {(response: Result<Movies, Error>) in
+            switch response {
+            case .failure(let error):
+                debugPrint(error)
+            case .success(let res):
+                self.movies[typeMovieSection] = res.movies
+            }
+            self.group.leave()
+        })
     }
     
     // MARK: - Helpers
+    private func configureUICollection() {
+        collectionView.register(HightSectionCell.self, forCellWithReuseIdentifier: HightSectionCell.reuseIdentifier)
+        collectionView.register(DefaultSectionCell.self, forCellWithReuseIdentifier: DefaultSectionCell.reuseIdentifier)
+        collectionView.register(TopRatedSectionCell.self, forCellWithReuseIdentifier: TopRatedSectionCell.reuseIdentifier)
+        collectionView.register(HomeHeader.self, forSupplementaryViewOfKind: categoryHomeHeaderId, withReuseIdentifier: HomeHeader.reuseIdentifier)
+    }
+    
     private func configureUI() {
-        view.backgroundColor = .systemBackground
+        navigationItem.title = "Movies"
+        view.backgroundColor = .black
     }
-    // MARK: - Actions
+    
+}
 
+// MARK: - UIControllerViewDataSource
+extension HomeViewController {
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let section = MovieGroupSections(rawValue: section) ?? .trending
+        return movies[section]?.count ?? 0
+    }
+    
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return MovieGroupSections.allCases.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let section = MovieGroupSections(rawValue: indexPath.section) ?? .topRated
+        guard let movies = movies[section] else { return DefaultSectionCell() }
+        let movie =  movies[indexPath.row]
+        
+        switch section {
+        case .popular:
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: HightSectionCell.reuseIdentifier, for: indexPath) as? HightSectionCell else {
```

Comment:
suggestion: https://github.com/AliSoftware/Reusable Take a look on how this works and you can implement it on your own